### PR TITLE
Dropped feature list, introduced protocol version, ...

### LINKED ISSUE #988: Drop feature list
Given each major releases provide their own batch of features, deprecation and such, it should become expected that the client no require to constantly look back on older legacy systems just to maintain a sense of backward compatibility.

In fact, some of the "features", if they are unsupported by the server, prevents clients from actively communicating with the server. So even that feature list code isn't properly maintained to keep compatibility with older servers despite the fact the entire selling point of the feature list IS TO DO SO.

### PR DESCRIPTION
**Pull request by TrickyLeifa**
Resolve #988

* Dropped feature list
* Introduced protocol version
* Bumped version to 2.11.0

### CODE REVIEW

**Comment by Salanto** [comment]
While it can join servers, it does not display the playercount anymore. I presume its the extra arg on the ID packet.
We might not want to break THAT specific aspect of Lobby just yet as it will make the client seem non-functional.
image

**Review by in1tiate — DISMISSED** [review]
Tentative approval pending review from @Salanto and testing

**Comment by stonedDiscord** [comment]
the playercount doesn't work

File: src/packet_distribution.cpp [hunk 1]
```diff
@@ -40,36 +40,33 @@ void AOApplication::server_packet_received(AOPacket packet)
...
-    AOPacket hi_packet("HI", f_contents);
-    send_server_packet(hi_packet);
-    log_to_demo = false;
+    QStringList f_contents = {get_protocol_version_string(), "AO2", get_version_string()};
+    send_server_packet(AOPacket("ID", f_contents));
   }
   else if (header == "ID")
   {
-    if (content.size() < 2)
+    if (content.size() < 3)
```
Comment (stonedDiscord) [inline src/packet_distribution.cpp:55]: i think this causes the playercount to stop working

**Review by stonedDiscord — CHANGES_REQUESTED** [review]

File: src/packet_distribution.cpp (hunk 1 above)
Comment (TrickyLeifa) [reply src/packet_distribution.cpp]: Protocol changes, intended.

**Review by Salanto — APPROVED** [review]

**Review by in1tiate — APPROVED** [review]

**Pull request closed without merging**